Tests/XCUITests/HistoryTests.swift

```diff
@@ -103,215 +113,264 @@ class HistoryTests: BaseTestCase {
-        navigator.openURL(path(forTestPage: "test-mozilla-book.html"))
-        waitUntilPageLoad()
-        waitForTabsButton()
-        navigator.goto(TabTray)
-        navigator.performAction(Action.AcceptRemovingAllTabs)
-        navigator.performAction(Action.CloseURLBarOpen)
+    func testRecentlyClosedWebsiteClosed() {
+        // Open "Book of Mozilla" and close the tab
+        openBookOfMozilla()
+        closeFirstTabByX()
+
+        // On regular mode, the closed tab is listed in "Recently Closed Tabs List"
         navigator.nowAt(NewTabScreen)
-        navigator.goto(BrowserTabMenu)
         navigator.goto(HistoryRecentlyClosed)
+        waitForExistence(app.tables["Recently Closed Tabs List"], timeout: TIMEOUT)
+        XCTAssertTrue(app.tables.cells.staticTexts[bookOfMozilla["label"]!].exists)
 
-        // The Closed Tabs list should contain the info of the website just closed
-        waitForExistence(app.tables["Recently Closed Tabs List"], timeout: 3)
-        XCTAssertTrue(app.tables.cells.staticTexts[closedWebPageLabel].exists)
-
-        navigator.goto(HomePanelsScreen)
-
-        // This option should be enabled on private mode too
+        // On private mode, the closed tab on regular mode is listed in "Recently Closed Tabs List" as well
         navigator.toggleOn(userState.isPrivate, withAction: Action.TogglePrivateMode)
         navigator.performAction(Action.OpenNewTabFromTabTray)
         navigator.nowAt(NewTabScreen)
-        navigator.performAction(Action.CloseURLBarOpen)
-        navigator.nowAt(NewTabScreen)
-        navigator.goto(BrowserTabMenu)
         navigator.goto(HistoryRecentlyClosed)
-        waitForExistence(app.tables["Recently Closed Tabs List"])
+        waitForExistence(app.tables["Recently Closed Tabs List"], timeout: TIMEOUT)
+        XCTAssertTrue(app.tables.cells.staticTexts[bookOfMozilla["label"]!].exists)
     }
 
-    func testClearRecentlyClosedHistory() {
-        // Open the default website
-        navigator.openURL(path(forTestPage: "test-mozilla-book.html"))
-        navigator.nowAt(BrowserTab)
+    func testRecentlyClosedPrivateMode() {
+        // Open "Book of Mozilla" on private mode and close the tab
+        navigator.toggleOn(userState.isPrivate, withAction: Action.TogglePrivateMode)
+        navigator.performAction(Action.OpenNewTabFromTabTray)
+        openBookOfMozilla()
+        closeFirstTabByX()
+
+        // "Recently Closed Tabs List" is empty
+        navigator.performAction(Action.OpenNewTabFromTabTray)
+        navigator.nowAt(NewTabScreen)
+        navigator.goto(LibraryPanel_History)
+        waitForExistence(app.tables[HistoryPanelA11y.tableView])
+        XCTAssertTrue(app.tables[HistoryPanelA11y.tableView].staticTexts[emptyRecentlyClosedMesg].exists)
+        XCTAssertFalse(app.tables.cells.staticTexts[bookOfMozilla["label"]!].exists)
+    }
+
+    func testRemoveAllTabsButton() {
```

Comment:
I would change the title of this test, we are not testing the remove all tabs button, but what happens when tabs are closed using it.
Do we have any answer on the issue you open? it may be a bug that using that button to close all tabs do not show them in recently closed. I would add a comment in the test about that issue and maybe comment it until we know the expected behaviour for sure.

Reply:
😞 No, I don't have an answer to FXIOS-5128.